File: Source/SaveSystem2.0/MAKE REWARDS HERE/SaveObjects/GameCurrency.cs
public class GameCurrency
{
    public KibbleCurrency Kibble { get; set; }

    public GameCurrency()
    {
        Kibble = new KibbleCurrency();
    }

    public void ResetGameCurrency()
    {
        Kibble.Reset();
    }
}

public class KibbleCurrency
{
    public int Collected { get; set; }
    public int Jackpots { get; set; }
    public int SuperJackpots { get; set; }

    public void Reset()
    {
        Collected = 0;
        Jackpots = 0;
        SuperJackpots = 0;
    }
}

File: src/SaveSystem2.0/MAKE REWARDS HERE/SaveObjects/GameCurrency.ts
export class GameCurrency {
    public Kibble: KibbleCurrency

    public constructor() {
        this.Kibble = new KibbleCurrency()
    }
}

export class KibbleCurrency {
    public Collected = 0
    public Jackpots = 0
    public SuperJackpots = 0
}
